gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -224,19 +225,19 @@ private List<IGitCoreReflogEntry> deriveReflogByRefFullName(String refFullName,
   }
 
   @Override
-  public Option<GitCoreRelativeCommitCount> deriveRelativeCommitCount(
+  public @Nullable GitCoreRelativeCommitCount deriveRelativeCommitCount(
       IGitCoreCommit fromPerspectiveOf,
       IGitCoreCommit asComparedTo) throws GitCoreException {
 
-    return withRevWalk(walk -> {
+    return (GitCoreRelativeCommitCount) withRevWalk(walk -> {
       val mergeBaseHash = deriveMergeBaseIfNeeded(fromPerspectiveOf, asComparedTo);
-      if (mergeBaseHash.isEmpty()) {
+      if (mergeBaseHash == null) {
         return Option.none();
```

Comment:
```suggestion
        return null;
```
Then remove `Option.some`, `.getOrNull()` and the explicit typecast

Reply:
oh, this part was really awkward. I wanted to ask, because the fullness checker, was getting confused about the ultimate type of the expression and throwing errors, when using `null`

Reply:
Should be okay if you apply the 4 changes listed above, I made it compile locally